test/test_vmap.py

```diff
@@ -3547,6 +3546,38 @@ def test_randperm(self, device, randomness, use_generator):
             for i in range(B0):
                 assert torch.allclose(vmap_result[i], expected)
 
+    @parametrize('randomness', ['error', 'same', 'different'])
+    def test_feature_dropout(self, device, randomness):
+        # special case because functional feature dropout expects a batch dimension, so it doesn't match the
+        # pattern for other randomness
+
+        seed = 1234567
+        supported_ops = [
+            lambda t: torch.nn.functional.dropout(torch.ones_like(t), training=True),
+            lambda t: torch.nn.functional.alpha_dropout(torch.ones_like(t), training=True),
+            lambda t: torch.nn.functional.feature_alpha_dropout(t, training=True),
+            lambda t: torch.nn.functional.dropout2d(t, training=True),
+            lambda t: torch.nn.functional.dropout3d(t.unsqueeze(-1), training=True),
+        ]
+
+        for op in supported_ops:
+            passed = torch.ones([4, 3, 3, 3, 3])
```

Comment:
Does this need `device=device`?

Reply:
Also, prefer to use the `B0` convention here so we know which dimension is being vmappd over
```
B0 = ...
passed = torch.ones(B0, 3, 3, 3, 3])
```